Network req data export improvements

hrajwade96 opened the pull request:
- Fixes https://github.com/flutter/devtools/issues/9421
- Additional fixes.


*Please add a note to `packages/devtools_app/release_notes/NEXT_RELEASE_NOTES.md` if your change requires release notes. Otherwise, add the 'release-notes-not-required' label to the PR.*

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or there is a reason for not adding tests.


![build.yaml badge]

If you need help, consider asking for help on [Discord].

<!-- Links -->
[Contributor Guide]: https://github.com/flutter/devtools/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[build.yaml badge]: https://github.com/flutter/devtools/actions/workflows/build.yaml/badge.svg

elliette commented on packages/devtools_app/lib/src/shared/http/http_request_data.dart:
```diff
@@ -101,15 +101,46 @@ class DartIOHttpRequestData extends NetworkRequest {
-        _responseBody = utf8.decode(fullRequest.responseBody!);
-        _requestBody = utf8.decode(fullRequest.requestBody!);
+        var responseMime =
+            responseHeaders?['content-type']?.toString().split(';').first;
```
Could we pull this into a helper function (e.g. `getHeadersMimeType`)? That way we can use it both here and below. Please add it to `network/utils/http_utils.dart` and add tests. Thanks!

hrajwade96 replied:
Added a helper function, will be adding the tests soon.

hrajwade96 commented on packages/devtools_app/lib/src/shared/http/http_request_data.dart:
```diff
@@ -101,15 +101,46 @@ class DartIOHttpRequestData extends NetworkRequest {
     }
   }
 
+  //TODO check if all cases are handled correctly
```
Done

hrajwade96 commented:
> > Hello @hrajwade96 - is this PR still on your radar or should it be closed? Thank you!
> 
> Hi @elliette, yes, I’m planning to finish this soon and will start making the remaining changes. Apologies for the gap in between!

@elliette I’ve addressed all the comments, could you please take another look at the PR?

hrajwade96 commented:
> Greetings from stale PR triage! 👋 Is this change still on your radar? Is this waiting on review from @elliette?

Hi @Piinks, yes this was pending for review, let me check and keep the PR upto date and synced with latest changes once.
